dds_web/api/project.py

```diff
@@ -44,161 +36,125 @@
+    )
+    return project
 
 
 class GetPublic(flask_restful.Resource):
     """Gets the public key beloning to the current project."""
 
-    method_decorators = [project_access_required, token_required]
-
-    def get(self, current_user, project):
+    @auth.login_required
+    def get(self):
         """Get public key from database."""
 
+        args = flask.request.args
+
+        project = verify(
+            current_user=auth.current_user(),
+            project_public_id=args.get("project"),
+            access_method=["get", "put"],
+        )
+
         app.logger.debug("Getting the public key.")
-        try:
-            proj_pub = (
-                models.Project.query.filter_by(public_id=project.get("id"))
-                .with_entities(models.Project.public_key)
-                .first()
-            )
 
-            if not proj_pub:
-                raise PublicKeyNotFoundError(project=project.get("id"))
+        if not project.public_key:
+            raise PublicKeyNotFoundError(project=project.public_id)
 
-        except sqlalchemy.exc.SQLAlchemyError as err:
-            raise DatabaseError(
-                message=str(err), username=current_user.username, project=project.get("id")
-            )
-        else:
-            return flask.jsonify({"public": proj_pub[0]})
+        return flask.jsonify({"public": project.public_key})
 
 
 class GetPrivate(flask_restful.Resource):
     """Gets the private key belonging to the current project."""
 
-    method_decorators = [project_access_required, token_required]
-
-    def get(self, _, project):
+    @auth.login_required
+    def get(self):
         """Get private key from database"""
 
+        args = flask.request.args
+
+        project = verify(
+            current_user=auth.current_user(),
+            project_public_id=args.get("project"),
+            access_method=["get"],
+        )
+
         # TODO (ina): Change handling of private key -- not secure
         app.logger.debug("Getting the private key.")
+
+        app_secret = app.config["SECRET_KEY"]
+        passphrase = app_secret.encode("utf-8")
+
+        enc_key = bytes.fromhex(project.private_key)
+        nonce = bytes.fromhex(project.privkey_nonce)
+        salt = bytes.fromhex(project.privkey_salt)
+
+        kdf = scrypt.Scrypt(
+            salt=salt,
+            length=32,
+            n=2 ** 14,
+            r=8,
+            p=1,
+            backend=backends.default_backend(),
+        )
+
+        key_enc_key = kdf.derive(passphrase)
         try:
-            proj_priv = (
-                models.Project.query.filter_by(public_id=project["id"])
-                .with_entities(
-                    models.Project.private_key,
-                    models.Project.privkey_nonce,
-                    models.Project.privkey_salt,
-                )
-                .first()
-            )
-        except sqlalchemy.exc.SQLAlchemyError as err:
```

Comment:
I think this is fine for now but we need to look into how to show what information to the end user.

Reply:
I will work on the keys and related parts in another branch later. It doesn't belong in the current work.

Reply:
I meant the part with `flask.make_response(str(err), 500)` below, that's what produces the codeql warning. But this is fine for now 👍🏻